File: encrypt.payload.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import base64, os, json

SECRET_KEY = os.getenv("BRIDGE_SECRET_KEY", "ThisIsA32ByteBridgeKey___").encode()

def encrypt_data(payload: dict) -> str:
    plaintext = json.dumps(payload).encode('utf-8')
    cipher = AES.new(SECRET_KEY, AES.MODE_ECB)
    ciphertext = cipher.encrypt(pad(plaintext, AES.block_size))
    return base64.b64encode(ciphertext).decode()

def decrypt_data(encoded: str) -> dict:
    cipher = AES.new(SECRET_KEY, AES.MODE_ECB)
    decrypted = unpad(cipher.decrypt(base64.b64decode(encoded)), AES.block_size)
    return json.loads(decrypted.decode())

if __name__ == "__main__":
    data = {
        "sender": "CordaBankA",
        "amount": "100000",
        "swiftCode": "MT103",
        "reference": "CDXFN-4234"
    }

    encrypted = encrypt_data(data)
    print("🔐 Encrypted Payload:", encrypted)

    with open("out/encrypted_payload.txt", "w") as f:
        f.write(encrypted)

File: encryptPayload.js
const crypto = require('crypto');
const fs = require('fs');

const key = Buffer.from(process.env.BRIDGE_SECRET_KEY || 'ThisIsA32ByteBridgeKey___');
const data = {
  sender: 'CordaBankA',
  amount: '100000',
  swiftCode: 'MT103',
  reference: 'CDXFN-4234'
};

const encryptPayload = (payload) => {
  const cipher = crypto.createCipheriv('aes-256-ecb', key, null);
  const encrypted = Buffer.concat([
    cipher.update(JSON.stringify(payload), 'utf8'),
    cipher.final()
  ]);
  return encrypted.toString('base64');
};

const encrypted = encryptPayload(data);
console.log('🔐 Encrypted:', encrypted);
fs.writeFileSync('out/encryptedPayload.txt', encrypted);
